src/main/java/org/openpnp/machine/reference/driver/SerialPortCommunications.java

```diff
@@ -137,8 +137,25 @@ public synchronized void disconnect() throws Exception {
     }
 
     public int read() throws TimeoutException, IOException {
-        byte[] b = new byte[1];
```

Comment:
cc @MatSpy looks like this change is from eb199d5e48a7905eea04cda748afb598b30233c2 

This is using jSerialComm.SerialPort bytesAvailable to check if there's new serial data (non-blocking) and then sleep the thread if not.. I suspect this was done to improve multi-threaded performance on the slower ND4 Atom CPU.

It looks like the main test branch is currently using setComPortTimeouts with TIMEOUT_READ_SEMI_BLOCKING, 500 ms read timeout which would cause this thread to block up to 500ms if no data is ready... although having that non-zero, longer (500ms) timeout might improve average latency to service incoming bytes (than yielding cycles to other threads)?

I haven't tested with/without this change personally. It could potentially impact timing behavior of other machines, so I'll go ahead and move it to a separate PR to make it cleaner to test against & report back on how removing this change impacts ND4.